### ISSUE TITLE
BinanceSocketClient has problems subscribing with SubscribeToPartialOrderBookUpdatesAsync to multiple streams

### ISSUE THREAD

**Issue by Khindasvinto** [JKorf/Binance.Net#1201]
Since 2023-01-30 if you make >= 2 subscriptions (3 if you connect to: data-stream.binance.com) with SubscribeToPartialOrderBookUpdatesAsync or >= 4 with SubscribeToTradeUpdatesAsync in a single socketClient 
you'll get: No response on subscription request received, on any new subscription that you try to make on that socketClient, so you have to create multiple socketClients when before a single one was enough.

You can't unsubscribe from this failed streams, the Data property is null and the cancellation token doesn't work either so you'll keep receiving unhandled messages.

Code:

	    var subscriptionPartialOrderBookBTC = await socketClient.SpotStreams.SubscribeToPartialOrderBookUpdatesAsync("BTCUSDT", 20, 100, data => { });

            if (!subscriptionPartialOrderBookBTC.Success)
                Console.WriteLine($"Failed BTC PartialOrderBook subscription: {subscriptionPartialOrderBookBTC.Error}");

            var subscriptionKlineBTC = await socketClient.SpotStreams.SubscribeToKlineUpdatesAsync("BTCUSDT", KlineInterval.OneMinute, data => { });

            if (!subscriptionKlineBTC.Success)
                Console.WriteLine($"Failed BTC Kline subscription: {subscriptionKlineBTC.Error}");

            var subscriptionPartialOrderBookETH = await socketClient.SpotStreams.SubscribeToPartialOrderBookUpdatesAsync("ETHUSDT", 20, 100, data => { });

            if (!subscriptionPartialOrderBookETH.Success)
                Console.WriteLine($"Failed ETH PartialOrderBook subscription: {subscriptionPartialOrderBookETH.Error}");

            var subscriptionKlineETH = await socketClient.SpotStreams.SubscribeToKlineUpdatesAsync("ETHUSDT", KlineInterval.OneMinute, data => { });            

            if (!subscriptionKlineETH.Success){
                Console.WriteLine($"Failed ETH Kline subscription. {subscriptionKlineETH.Error}");

                if (subscriptionKlineETH.Data == null)
                    Console.WriteLine("subscriptionKlineETH.Data is null");
            }

Log:

[10:57:54 DBG] Binance    | Socket 1 new socket created for wss://stream.binance.com:9443/stream
[10:57:54 DBG] Binance    | Socket 1 adding new subscription with id 2, total subscriptions on connection: 1
[10:57:54 DBG] Binance    | Socket 1 connecting
[10:57:57 DBG] Binance    | Socket 1 connected to wss://stream.binance.com:9443/stream
[10:57:57 DBG] Binance    | Socket 1 status changed from None to Connected
[10:57:57 DBG] Binance    | Socket 1 starting processing tasks
[10:57:57 INF] Binance    | Socket 1 subscription 2 completed successfully
[10:57:57 DBG] Binance    | Socket 1 adding new subscription with id 4, total subscriptions on connection: 2
[10:57:59 INF] Binance    | Socket 1 subscription 4 completed successfully
[10:57:59 DBG] Binance    | Socket 1 adding new subscription with id 6, total subscriptions on connection: 3
[10:58:04 INF] Binance    | Socket 1 subscription 6 completed successfully
[10:58:04 DBG] Binance    | Socket 1 adding new subscription with id 8, total subscriptions on connection: 4
[10:58:14 WRN] Binance    | Socket 1 failed to subscribe: : No response on subscription request received
[10:58:14 DBG] Binance    | Socket 1 closing subscription 8
Failed ETH Kline subscription. : No response on subscription request received
subscriptionKlineETH.Data is null
[10:58:30 WRN] Binance    | Socket 1 Message not handled: {
  "result": null,
  "id": 7
}
[10:58:45 WRN] Binance    | Socket 1 Message not handled: {
  "stream": "ethusdt@kline_1m",
  "data": {
    "e": "kline",
    "E": 1676109487033,
    "s": "ETHUSDT",
    "k": {
      "t": 1676109480000,
      "T": 1676109539999,
      "s": "ETHUSDT",
      "i": "1m",
      "f": 1075723634,
      "L": 1075723664,
      "o": "1522.10000000",
      "c": "1522.10000000",
      "h": "1522.11000000",
      "l": "1522.10000000",
      "v": "0.74910000",
      "n": 31,
      "x": false,
      "q": "1140.20934200",
      "V": "0.42320000",
      "Q": "644.15695200",
      "B": "0"
    }
  }
}

**Comment by JKorf** [comment]
Hi, I can't reproduce this with the code you provided, however I did add functionality to unsubscribe the stream again if the request has timed out, but a confirmation comes in later. So that should at least fix the issue of the data coming in but not getting handled and no way to unsubscribe.

**Comment by Khindasvinto** [comment]
The problem with the unhandled messages has been resolved, thank you very much. 

But I'm still unable to connect to more than 4 streams (if I use SubscribeToTradeUpdatesAsync or SubscribeToPartialOrderBookUpdatesAsync) in a single socketClient, it seems that with every new subscription the response time of the socketClient gets degraded. 

Using the same code of my first message I changed the SocketResponseTimeout to 180 seconds and only in around the 20% of occassions I get a successful connection to the last stream, and when I get it is after many seconds of waiting:

[19:53:32 DBG] Binance    | Socket 1 new socket created for wss://stream.binance.com:9443/stream
[19:53:32 DBG] Binance    | Socket 1 adding new subscription with id 2, total subscriptions on connection: 1
[19:53:32 DBG] Binance    | Socket 1 connecting
[19:53:34 DBG] Binance    | Socket 1 connected to wss://stream.binance.com:9443/stream
[19:53:34 DBG] Binance    | Socket 1 status changed from None to Connected
[19:53:34 DBG] Binance    | Socket 1 starting processing tasks
[19:53:34 INF] Binance    | Socket 1 subscription 2 completed successfully
[19:53:34 DBG] Binance    | Socket 1 adding new subscription with id 4, total subscriptions on connection: 2
[19:53:35 INF] Binance    | Socket 1 subscription 4 completed successfully
[19:53:35 DBG] Binance    | Socket 1 adding new subscription with id 6, total subscriptions on connection: 3
[19:53:47 INF] Binance    | Socket 1 subscription 6 completed successfully
[19:53:47 DBG] Binance    | Socket 1 adding new subscription with id 8, total subscriptions on connection: 4
[19:55:42 INF] Binance    | Socket 1 subscription 8 completed successfully

If I create a socketClient for every stream using SubscribeToTradeUpdatesAsync or SubscribeToPartialOrderBookUpdatesAsync I don't get problems and can subscribe to many streams so I think it's not a problem of my connection or Binance's fault. I also don't have any problems subscribing to many kline streams (with SubscribeToKlineUpdatesAsync) in a single socketClient.
